### PR TITLE
2.x: Improve prefix default (for caches etc.)

### PR DESCRIPTION
**Pull request by ionas**
See: https://github.com/cakephp/cakephp/issues/5645 for discussion.

### CODE REVIEW

**Comment by ADmad** [comment]
The app path could be potentially quite long and using such long prefixes as default would be silly.

**Comment by lorenzo** [comment]
This how it was before, and it caused unnoticed bugs because people thought that calling Inflector was an important thing to do so they did not change the prefix. At least now it looks like something you can change.

**Comment by ionas** [comment]
@ADmad yes it could be a long prefix. I can't see however, why that is silly and if it has any major negative impact.

For me the way it is currently caused bugs, had it been the way purposed in the PR I would have not gotten into troubles.

**Comment by ionas** [comment]
Also I am not sure if before it wasnt APP_DIR which is totally different from APP.

**Comment by AD7six** [comment]
:-1: 

Users are [already told to make sure it's unique](https://github.com/cakephp/cakephp/pull/5647/files#diff-c0f895b4c5ba74a84f8a54a144adbfeaR361). This adds calls to `Inflector::slug` for each request and has the added problem of misleading devs to think it's not something they should edit.

Do we really want:

    app/tmp/cache/persistent/
        var_www_domain_dev_htdocs_appobject_map
        var_www_domain_dev_htdocs_appcake_core_cake_en
        ...

That might be unique but it makes the prefix hard to read by humans.

**Comment by dakota** [comment]
:-1: as stated by @AD7six, it is already documented as needing to be unique.

**Comment by ADmad** [comment]
I think there are enough :-1: and the reason for it to close this.

**Comment by ionas** [comment]
Having conventions in work is arguably better than having to configure things.
- You can always edit away Inflector::slug(). This is how convention over configuration works.
- You can always add a comment "edit at free will but make sure it is unique per APC instance".
- I am fine with those names, it is just cache names. Unique and meaningful cache names.

Its a community project and you are the core but I think you are wrong. The approach before was the better once, just it should have been APP_DIR or any other semi-unique string that works out of the box by convention.

But it seems more configuration is a great thing, especially error prone - just lovely.

p.s.: I am not suggesting this cause I like to know better but because I ran into production issues with this, and I would have probably overlooked it in any way - be it _myapp or Inflector::slug(APP) - which it was before. Inflector::slug(APP_DIR) looks like it should not be changed while it should - but _myapp looks nothing different. Inflector::slug(APP) will simply work wonderfully for all those who don't care to optimize for speed.

You can, and you can still afterwards, it simply won't be optimized for speed out of the box but for no apc clashes.

**Comment by lorenzo** [comment]
@ionas I think the same argument can be applied to the current code: you can always make it unique in anyway you like, which could include using Inflector::slug().

The comment is already there, it is important to make it unique. I guess the reason you ran into the same problem is because you missed reading that same comment.

If there were a way to make it more obvious to people that the cache key is not unique, I would prefer that. Kind of like the warnings we have for Security.salt

**Comment by AD7six** [comment]
@lorenzo I like that idea.

Would also be handy to have a built in cli interface for these kind of checks.

**Comment by ionas** [comment]
If you want to keep the preference of speed + configuration over convention :/ out of the box then I'd suggest something like $prefix = 'CHANGE_THIS_TO_A_UNIQUE_FILE_PREFIX_FOR_APC_AND_MEMCACHE'; // yes it is ugly; which is the point.
'_myApp' looks pretty generic and thus looks like "should work out of the box".
Hopefully there is some better approach in 3.0 (did not look)

**Comment by dereuromark** [comment]
I agree with @lorenzo and @AD7six . The core.php template shows how you can make it unique, any way you like:

    // Prefix each application on the same server with a different string, to avoid ...

I find that very easy to understand actually. No need to further clarify it beyond that.
But a warning/check sounds good to me, too.

**Comment by tersmitten** [comment]
What about a warning in `app/View/Pages/home.ctp`?

File: app/Config/core.php [hunk 1]
```diff
@@ -359,7 +359,7 @@
 }
 
 // Prefix each application on the same server with a different string, to avoid Memcache and APC conflicts.
-$prefix = 'myapp_';
+$prefix = Inflector::slug(APP);
```
Comment (adel1213) [inline app/Config/core.php:362]: app

Comment (dereuromark) [reply app/Config/core.php:362]: Please dont comment on a 5+ year old ticket and instead verify this on the current branches and open an issue or PR where needed instead.

**Review by adel1213 — COMMENTED** [review]
App

**Pull request closed without merging**